src/functions/merge.js

```diff
@@ -1,5 +1,9 @@
 'use strict';
 
+function isPlainObject(value) {
```

Comment:
This test is actually called `isObjectLike` in lodash. I'm think it might not be a bad idea to keep lodash naming when the implem is the same. I'll let you the judge of that.
https://github.com/lodash/lodash/blob/master/isObjectLike.js#L23